doordeck-sdk/src/mingwMain/resources/doordeck_headless_sdk/DoordeckSdk.cs

```diff
@@ -0,0 +1,131 @@
+    private readonly Accountless _accountless = new();
+    private readonly Fusion _fusion = new();
+    private readonly Helper _helper = new();
+    private readonly LockOperations _lockOperations = new();
+    private readonly Platform _platform = new();
+    private readonly Sites _sites = new();
+    private readonly Tiles _tiles = new();
+    private readonly ContextManager _contextManager = new();
+    private readonly CryptoManager _cryptoManager = new();
+
+    public void Initialize()
+    {
+        var token = authToken != null ? Utils.Utils.ToSByte(authToken) : null;
+        _apiEnvironment = apiEnvironment switch
+        {
+            ApiEnvironment.DEV => _symbols->kotlin.root.com.doordeck.multiplatform.sdk.api.model.ApiEnvironment.DEV
+                .get(),
+            ApiEnvironment.STAGING => _symbols->kotlin.root.com.doordeck.multiplatform.sdk.api.model.ApiEnvironment
+                .STAGING.get(),
+            ApiEnvironment.PROD => _symbols->kotlin.root.com.doordeck.multiplatform.sdk.api.model.ApiEnvironment.PROD
+                .get(),
+            _ => _apiEnvironment
+        };
+
+        _factory = _symbols->kotlin.root.com.doordeck.multiplatform.sdk.KDoordeckFactory._instance();
+
+        if (authToken != null)
+        {
+            _sdk = _symbols->kotlin.root.com.doordeck.multiplatform.sdk.KDoordeckFactory.initializeWithAuthToken(_factory, _apiEnvironment, token);
+            Marshal.FreeHGlobal((IntPtr)token);
+        }
+        else
+        {
+            _sdk = _symbols->kotlin.root.com.doordeck.multiplatform.sdk.KDoordeckFactory.initialize(_factory, _apiEnvironment);
+        }
+
+        _account.Initialize(_symbols, _sdk);
+        _accountless.Initialize(_symbols, _sdk);
+        _fusion.Initialize(_symbols, _sdk);
+        _helper.Initialize(_symbols, _sdk);
+        _lockOperations.Initialize(_symbols, _sdk);
+        _platform.Initialize(_symbols, _sdk);
+        _sites.Initialize(_symbols, _sdk);
+        _tiles.Initialize(_symbols, _sdk);
+        _contextManager.Initialize(_symbols, _sdk);
+        _cryptoManager.Initialize(_symbols, _sdk);
+    }
+
+    public Account GetAccount()
+    {
+        return _account;
+    }
+
+    public Accountless GetAccountless()
+    {
+        return _accountless;
+    }
+
+    public Fusion GetFusion()
+    {
+        return _fusion;
+    }
+
+    public Helper GetHelper()
+    {
+        return _helper;
+    }
+
+    public LockOperations GetLockOperations()
+    {
+        return _lockOperations;
+    }
+
+    public Platform GetPlatform()
+    {
+        return _platform;
+    }
+
+    public Sites GetSites()
+    {
+        return _sites;
+    }
+
+    public Tiles GetTiles()
+    {
+        return _tiles;
+    }
+
+    public ContextManager GetContextManager()
+    {
+        return _contextManager;
+    }
+
+    public CryptoManager GetCryptoManager()
+    {
+        return _cryptoManager;
+    }
+
+    public void Release()
```

Comment:
AI suggests:

	•	The Release method relies on correct manual disposal via DisposeStablePointer and other calls. This makes the class highly prone to memory leaks. Consider implementing IDisposable and using patterns like using for resource management.
	•	Are you certain all pinned objects (pinned) are released correctly in all failure scenarios? You might need additional error-handling and testing for resource leaks.

Reply:
This release method is intended to be called manually at the application exit. Right now, I don't see how to improve it, but I will keep an eye on it!